code/script.js

```diff
@@ -247,53 +251,68 @@ const generateBoard = () => {
+  console.log(secret)
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  // The board will be visible by adding GenerateBoard below
+  generateBoard() 
+  // Also the code will select one randomly person by adding setSecret function below
+  setSecret() 
+  board.style.display="flex" // 
+  winOrLose.style.display="none"
+
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// stores one value when clicking to the dropdown and selects/clicking one option
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const value = questions.options[questions.selectedIndex].value
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: value,
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category,}
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: value,
+      value: true, 
       category: category,
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true, 
+      category: category,
+    }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// This function is invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  if (secret[currentQuestion.attribute] === currentQuestion.value) { 
+    filterCharacters(true)
+  } else {
+    filterCharacters(false)
+  }
 }
 
-// It'll filter the characters array and redraw the game board.
+// This function will filter the characters array and redraw the game board after we selected one value from dropdown and pressed 'find-out'.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const attribute = currentQuestion.attribute;
+  const value = currentQuestion.value;
+  const category = currentQuestion.category;
```

Comment:
You should avoid global variables as long as possible. It is not big harm here, but you should train your brain to look for easy ways to avoid it. 

one trick I can show here is the object deconstruction: `const {attribute, value, category } = currentQuestion;`